hax/test/integration/test_motr.py

```diff
@@ -71,6 +72,77 @@ def fn(trace: Invocation) -> bool:
     return fn
 
 
+def node_failed() -> TraceMatcher:
+    '''
+    Returns true if the second argument is a array to pointer to HaNoteStruct
+    and it contains 5 elements and that structure brings M0_NC_FAILED state.
+    In case of node failure we should see 5 elements(1 IO service, 1 node fid,
+    1 encl fid, 2 ctrls) in array.
+
+    It makes sense to use this matcher together with
+    `tr_method('ha_broadcast')`
+    '''
+    def fn(trace: Invocation) -> bool:
+        if len(trace.args) < 2:
+            return False
+        note = trace.args[1]
+        if (not isinstance(note, ctypes.Array) or
+                len(note) != 5):
+            return False
+
+        fids = [FidStruct(0x7200000000000001, 0x15),
```

Comment:
The matcher code implicitly requires the order in which Fids appear in the HA notes. Do we really want to make the order that hard?

Reply:
Done, modified code to remove dependency on order